sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1159,3 +1161,117 @@ def test_large_regularization():
         with np.errstate(all='raise'):
             model.fit(iris.data, iris.target)
         assert_array_almost_equal(model.coef_, np.zeros_like(model.coef_))
+
+
+def _test_gradient_common(loss_function, cases):
```

Comment:
I think the tests would be more robust and more readable if you use finite difference:

```expected = (loss_function.loss(p + epsilon) - loss_function.loss(p - epsilon)) / (2 * epsilon)```